### ISSUE TITLE
Username is not updated along with email in non-activated coauthor acct; can result in errors if email address is reused before coauthor acct is activated (and username changed)

### ISSUE THREAD

**Issue by alainna** [BirkbeckCTP/janeway#3606]
**Describe the bug**
If a Submitting Author/Manager makes an error inputting the email address of a preprint Coauthor w/o a Janeway account during submission, the erronious email address remains associated with the inactive Coauthor account as a username and is not updated if the email address is changed by Submitting Author/Moderator at a later point. 

This becomes especially problematic if Submitting Author accidentally puts Coauthor C's email address in Coauthor B's author entry and neglects to add Coauthor C. When Manager edits author list to correct Coauthor B's address, Coauthor C's email remains as the username. When Manager tries to add Coauthor C, they receive an error: the address already in the system. When Manager trii to use search by email to add Coauthor C: address not found. 

**Janeway version**
1.4+ Repository 

**To Reproduce**
1. Submit a preprint with one Coauthor (Author A), using an incorrect email address
![image](https://github.com/BirkbeckCTP/janeway/assets/15632391/1d62abfa-167d-4506-8b29-1d33d6861e7f)

2. Update the email address for Author A to the correct address 
![image](https://github.com/BirkbeckCTP/janeway/assets/15632391/d9efbc5a-93e4-4a09-aa5f-7f59d27ae708)

3. Confirm Author A's account details in admin view
![image](https://github.com/BirkbeckCTP/janeway/assets/15632391/c8dfae3b-152d-477c-89f8-de590db0702e)

4. Add new Coauthor (Author B) with the incorrect address used in step 1
5. See error
![image](https://github.com/BirkbeckCTP/janeway/assets/15632391/a259a51b-215d-4863-a8d9-36f91f9e1463)

**Expected behavior**
For an inactive account, the username would be updated along with email address. Alternatively: username would not exist until account is activated.

**Comment by ajrbyers** [comment]
It really annoys me that Django doesn't handle Duplicate errors well.

**Comment by joemull** [comment]
@S-Haime I don't believe it does, but it's also possible we have fixed it elsewhere. I think we need to investigate it to try to replicate it.

**Comment by S-Haime** [comment]
Superceded by change to frozen authors.
